mlflow/store/abstract_store.py

```diff
@@ -50,6 +50,16 @@ def get_experiment(self, experiment_id):
         """
         pass
 
+    @abstractmethod
+    def get_experiment_by_name(self, experiment_name):
```

Comment:
It looks like all three implementations of this method invoke `list_experiments()` and match against the provided experiment name. That seems to suggest that we can provide an implementation of the method in the base class. If this implementation is not performant enough for a given use case, the implementor can always override the method.

I noticed that the `file_store` implementation makes an additional call to `_check_root_dir` prior to calling `list_experiments`. However, `list_experiments` immediately calls `_check_root_dir`, so this call seems to be superfluous: all three implementations seem to share the same structure.

Reply:
Like your idea of base implementation or adding specific implementations for each. 

Will take care of this in a follow on change since this is a PR for `--experiment-name` change (which needed this `get_experiment_by_name` method for completeness)